src/main/java/ai/datahunters/md/server/photos/upload/ArchiveHandler.java

```diff
@@ -0,0 +1,69 @@
+package ai.datahunters.md.server.photos.upload;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.tika.Tika;
+import org.apache.tika.metadata.Metadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class ArchiveHandler {
+    public static final String TAR="application/x-gtar";
+    public static final String ZIP="application/zip";
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public Mono<List<String>> unarchive(ArchiveInputStream archive) {
+        List<String> files = new ArrayList<>();
+        try {
+            ArchiveEntry entry;
+            logger.info(String.valueOf(archive.getBytesRead()));
+            while ((entry = archive.getNextEntry()) != null) {
+                if (entry.isDirectory()) {
```

Comment:
for now we can assume that we are dealing with zips with files only, without any directories inside. In case there is something we cannot handle now - throw